### PR TITLE
Link back to likely declarations in "did you mean...?" error messages

### LINKED ISSUE #25031: Related error span for "did you mean ...?" error messages
Now that we support multiple related spans for errors (#10489, #22789, #24548), we'd like to improve an existing error message.

Currently, we provide an error when accessing declarations that have been misspelled.

```
Property '{0}' does not exist on type '{1}'. Did you mean '{2}'?
Cannot find name '{0}'. Did you mean '{1}'?
Object literal may only specify known properties, but '{0}' does not exist in type '{1}'. Did you mean to write '{2}'?
Cannot find name '{0}'. Did you mean the static member '{1}.{0}'?
Cannot find name '{0}'. Did you mean the instance member 'this.{0}'?
Module '{0}' has no exported member '{1}'. Did you mean '{2}'?
```

We can add a related span on the declaration of the suggestion if one exists:

```
It's likely that you meant to reference '{0}'.
```

This can help users quickly glance at the original declaration.

### LINKED ISSUE #25473: Related spans *probably* should all be 'message'?
Right now `7038` and `2728` use `"Error"` as their kind; but other related spans us `"Message"`

### LINKED ISSUE #25475: Rename 'getSuggestionForNonexistentModule' in 'TypeChecker'
It should be called `getSuggestionForNonexistentExport`.

### PR DESCRIPTION
**Pull request by DanielRosenwasser**
This pull request provides a related span on the declaration we're suggesting in our "did you mean...?" error messages.

thingyoumeantwasdeclaredhere

It currently doesn't handle union/intersection properties, or some corrections like with instance/static properties. So I guess it's a work in progress depending on how deep we want to go.

Fixes #25031

Also:

Fixes #25473
Fixes #25475

### CODE REVIEW

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -17757,11 +17775,20 @@ namespace ts {
                 // However, resolveNameHelper will continue and call this callback again, so we'll eventually get a correct suggestion.
                 return symbol || getSpellingSuggestionForName(unescapeLeadingUnderscores(name), arrayFrom(symbols.values()), meaning);
             });
-            return result && symbolName(result);
+            return result;
+        }
+
+        function getSuggestionForNonexistentSymbol(location: Node | undefined, outerName: __String, meaning: SymbolFlags): string | undefined {
```
Comment (mhegazy) [inline src/compiler/checker.ts]: we should have made these all internal :(

Comment (DanielRosenwasser) [reply src/compiler/checker.ts]: Yeah. 😡

We still can?

Comment (mhegazy) [reply src/compiler/checker.ts]: it is a breaking change..

Comment (mhegazy) [reply src/compiler/checker.ts]: but yes we can

**Pull request merged**